File: blossomedu_client/lib/core/models/announcement.dart
class Announcement {
  final int id;
  final String title;
  final String content;
  final String? image;
  final String authorName;
  final String createdAt;
  final bool isActive;

  Announcement({
    required this.id,
    required this.title,
    required this.content,
    this.image,
    required this.authorName,
    required this.createdAt,
    required this.isActive,
  });

  factory Announcement.fromJson(Map<String, dynamic> json) {
    return Announcement(
      id: json['id'],
      title: json['title'],
      content: json['content'],
      image: json['image'],
      authorName: json['author_name'] ?? 'Unknown',
      createdAt: json['created_at'],
      isActive: json['is_active'] ?? true,
    );
  }
}

File: core/models/announcement.py
from django.db import models
from django.conf import settings

class Announcement(models.Model):
    title = models.CharField(max_length=200)
    content = models.TextField()
    image = models.ImageField(upload_to='announcements/', null=True, blank=True)
    author = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
    created_at = models.DateTimeField(auto_now_add=True)
    is_active = models.BooleanField(default=True)

    def __str__(self):
        return self.title
